client.go

```diff
@@ -326,19 +342,19 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 
 	response := &Response{Response: resp}
 
-	err = CheckResponse(resp)
+	ok, err := CheckResponse(resp, errorResponse)
```

Comment:
```suggestion
	err := CheckResponse(resp, errorResponse)
```
Acho que aqui a gente pode retornar só o erro, sem o booleano